sympy/plotting/plot.py

```diff
@@ -1490,6 +1491,10 @@ def _process_series(self, series, ax, parent):
         if parent.legend:
             if ax.legend():
                 ax.legend_.set_visible(parent.legend)
+        if parent.legend_loc:
+            if ax.legend():
+                _loc = self.matplotlib.legend.Legend.codes.get(parent.legend_loc, 'best')
+                ax.legend_._set_loc(_loc)
```

Comment:
> This looks like using a private function of matplotlib, which can be very fragile Is there any way to use public function to update this?

Unfortunately, there is no proper public function to update the `_loc` attribute after the `Legend` object has been created. 
Maybe direct assignment is a better way? 
```
ax.legend_._loc = _loc
```

Reply:
> Maybe there can be an issue raised for matplotlib because they would need to support the imperative commands for that as well

Thanks, I'll try to open an issue for this problem.

Reply:
Reference: https://github.com/matplotlib/matplotlib/issues/25014
It seems that matplotlib does not intend to support public functions of legends. 
Should this pr be turned off?